drive/src/contract/mod.rs

```diff
@@ -267,16 +323,34 @@ impl DocumentType {
     pub fn from_cbor_value(
         name: &str,
         document_type_value_map: &[(Value, Value)],
+        default_keeps_history: bool,
+        default_mutability: bool,
     ) -> Result<Self, Error> {
         let mut document_properties: HashMap<String, types::DocumentFieldType> = HashMap::new();
 
+        // Do documents of this type keep history? (Overrides contract value)
+        let documents_keep_history: bool = cbor_inner_bool_value_with_default(
+            document_type_value_map,
+            "documentsKeepHistory",
+            default_keeps_history,
+        );
+
+        // Are documents of this type mutable? (Overrides contract value)
+        let documents_mutable: bool = cbor_inner_bool_value_with_default(
+            document_type_value_map,
+            "documentsMutable",
```

Comment:
"readOnly"

Reply:
From the docs ```If "readOnly" has a value of boolean true, it indicates that the value of the instance is managed exclusively by the owning authority, and attempts by an application to modify the value of this property are expected to be ignored or rejected by that owning authority```

Basically if you put readOnly instead of documentsMutable, it would mean that the documentType instance definition can not be changed other than the owning authority, not that instances from that documentType could not be changed.